cdc/redo/manager.go

```diff
@@ -349,6 +364,16 @@ func (m *ManagerImpl) UpdateResolvedTs(
 	return nil
 }
 
+// UpdateCheckpointTs updates checkpoint to the manager.
+func (m *ManagerImpl) UpdateCheckpointTs(ckpt model.Ts) {
+	for {
+		curr := atomic.LoadUint64(&m.checkpointTs)
+		if ckpt <= curr || atomic.CompareAndSwapUint64(&m.checkpointTs, curr, ckpt) {
```

Comment:
When will `ckpt <= curr` happen?

Reply:
It happends when the function is called in multiple threads.

Reply:
if global checkpoint is set here,, this condition will not happen?